static/js/market.js

```diff
@@ -0,0 +1,340 @@
+const market = async () => {
+  Vue.component(VueQrcode.name, VueQrcode)
+
+  const NostrTools = window.NostrTools
+
+  const defaultRelays = [
+    'wss://relay.damus.io',
+    'wss://relay.snort.social',
+    'wss://nostr-pub.wellorder.net',
+    'wss://nostr.zebedee.cloud'
+  ]
+  const eventToObj = event => {
+    event.content = JSON.parse(event.content)
+    return {
+      ...event,
+      ...Object.values(event.tags).reduce((acc, tag) => {
+        let [key, value] = tag
+        return {...acc, [key]: [...(acc[key] || []), value]}
+      }, {})
+    }
+  }
+  await Promise.all([
+    productCard('static/components/product-card/product-card.html'),
+    customerMarket('static/components/customer-market/customer-market.html'),
+    customerStall('static/components/customer-stall/customer-stall.html'),
+    productDetail('static/components/product-detail/product-detail.html'),
+    shoppingCart('static/components/shopping-cart/shopping-cart.html'),
+    chatDialog('static/components/chat-dialog/chat-dialog.html')
+  ])
+
+  new Vue({
+    el: '#vue',
+    mixins: [windowMixin],
+    data: function () {
+      return {
+        account: null,
+        accountDialog: {
+          show: false,
+          data: {
+            watchOnly: false,
+            key: null
+          }
+        },
+        drawer: false,
+        pubkeys: new Set(),
+        relays: new Set(),
+        events: [],
+        stalls: [],
+        products: [],
+        profiles: new Map(),
+        searchText: null,
+        inputPubkey: null,
+        inputRelay: null,
+        activePage: 'market',
+        activeStall: null,
+        activeProduct: null
+      }
+    },
+    computed: {
+      filterProducts() {
+        let products = this.products
+        if (this.activeStall) {
+          products = products.filter(p => p.stall_id == this.activeStall)
+        }
+        if (!this.searchText || this.searchText.length < 2) return products
+        return products.filter(p => {
+          return (
+            p.name.includes(this.searchText) ||
+            p.description.includes(this.searchText) ||
+            p.categories.includes(this.searchText)
+          )
+        })
+      },
+      stallName() {
+        return this.stalls.find(s => s.id == this.activeStall)?.name || 'Stall'
```

Comment:
- maybe just have `this.activeStall` be a stall object
- then use `this.activeStall.id`, `{{activeStall.name}}`, etc

Reply:
When i set these variables, activeStall, activeProduct, etc... there's no stalls or products yet! it just parses them from the query params. Also used to set query params when changing pages! Let's keep it for now and revisit it later, is that ok?

Reply:
got it, ok!